### PR TITLE
"osc shell" uses --noinit by default now if buildroot exists

### PR DESCRIPTION
**Pull request by adrianschroeter**

### CODE REVIEW

File: osc/build.py [hunk 1]
```diff
@@ -866,13 +869,14 @@ def main(apiurl, opts, argv):
     # real arch of this machine
     # vs.
     # arch we are supposed to build for
-    if bi.hostarch != None:
```
Comment (darix) [inline osc/build.py]: is this an unrelated change?

**Comment by lethliel** [comment]
@adrianschroeter Please open separate PRs for the commits. As they have nothing in common.

File: osc/build.py [hunk 2]
```diff
@@ -675,6 +673,11 @@ def main(apiurl, opts, argv):
         except:
             pass
 
+    if opts.shell:
+        buildargs.append("--shell")
+        if os.path.exists(build_root) and not opts.clean:
+            buildargs.append("--noinit")
```
Comment (lethliel) [inline osc/build.py]: opts.noinit must be set to True (otherwise the buildinfo and the buildconfig will be downloaded from the server again.)
opts.offline is also needed if the build_root exists.

Comment (adrianschroeter) [reply osc/build.py]: done

**Comment by lethliel** [comment]
Thanks for the patch. And LGTM
